paasta_tools/check_oom_events.py

```diff
@@ -0,0 +1,146 @@
+    ))
+    parser.add_argument(
+        '-d', '--soa-dir', dest="soa_dir", default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    parser.add_argument(
+        '-r', '--realert-every', dest="realert_every", type=int, default=1,
+        help="Sensu 'realert_every' to use.",
+    )
+    parser.add_argument(
+        '-s', '--superregion', dest="superregion", required=True,
+        help="The superregion to read OOM events from.",
+    )
+    return parser.parse_args(args)
+
+
+def oom_events(cluster, superregion, num_lines=1000):
+    """Read the latest 'num_lines' lines from OOM_EVENTS_STREAM and iterate over them."""
+    host_port = choice(scribereader.get_default_scribe_hosts(tail=True))
+    stream = scribereader.get_stream_tailer(
+        OOM_EVENTS_STREAM, host_port['host'], host_port['port'], True,
+        num_lines, superregion=superregion,
+    )
+    for line in stream:
+        try:
+            j = json.loads(line)
+            if j.get('cluster', '') == cluster:
+                yield j
+        except json.decoder.JSONDecodeError:
+            pass
+
+
+def latest_oom_events(cluster, superregion, interval=60):
+    """
+    :returns: {(service, instance): [(hostname, container_id, process_name), ...] }
+              if the number of events > 0
+    """
+    start_timestamp = int(time.time()) - interval
+    res = {}
+    for e in oom_events(cluster, superregion):
+        if e['timestamp'] > start_timestamp:
+            key = (e['service'], e['instance'])
+            res.setdefault(key, []).append((
+                e.get('hostname', ''),
+                e.get('container_id', ''),
+                e.get('process_name', ''),
+            ))
+    return res
+
+
+def compose_sensu_status(service_instance, events):
+    """
+    :param service_instance: a tuple (service, instance)
+    :param events: a list of tuples (hostname, container_id, process_name)
+    """
+    if len(events) == 0:
+        return (Status.OK, 'oom-killer is calm.')
+    elif len(events) == 1:
+        return (Status.CRITICAL, 'killing 1 process/min (%s).' % events[0][2])
+    else:
+        return (
+            Status.CRITICAL, 'killing %d processes/min (%s).' %
+            (len(events), ','.join(sorted(e[2] for e in events if e[2]))),
+        )
+
+
+def send_sensu_event(instance, status, args):
+    check_name = compose_check_name_for_service_instance(
+        'oom-killer',
+        instance.service,
+        instance.instance,
+    )
+    monitoring_overrides = instance.get_monitoring()
+    monitoring_overrides['page'] = False
+    monitoring_overrides['ticket'] = False
+    monitoring_overrides['alert_after'] = '1m'
+    monitoring_overrides['runbook'] = ['http://y/none']  # TODO: needs a link
+    monitoring_overrides['realert_every'] = args.realert_every
+    monitoring_overrides['tip'] = 'Increase memory limit.'
+    if monitoring_overrides['team'] == 'operations':  # TODO: remove after initial testing
```

Comment:
Can you send all alerts to yourself instead by overriding the team directly? (to noop)